kore/app/share/GlobalMain.hs

```diff
@@ -299,16 +309,18 @@ globalCommandLineParser =
         (  long "version"
         <> help "Print version information" )
 
+newtype ForKoreExec = ForKoreExec { used :: Bool }
 
 -- | Run argument parser for local executable
 commandLineParse
-    :: Parser a                -- ^ local options parser
+    :: ForKoreExec
+    -> Parser a                -- ^ local options parser
     -> InfoMod (MainOptions a) -- ^ local parser info modifiers
     -> IO (MainOptions a)
-commandLineParse localCommandLineParser modifiers = do
+commandLineParse (ForKoreExec used) localCommandLineParser modifiers = do
```

Comment:
We should use [`ExeName`](https://github.com/kframework/kore/blob/master/kore/src/Kore/Log/KoreLogOptions.hs#L203) here instead, so that we can still give a helpful message when not using `kore-exec`.